### PR TITLE
try to check emails with params

### LINKED ISSUE #49: Emails with params
Example:

```
 someone@example.com?subject=This%20is%20the%20subject&cc=someone_else@example.com&body=This%20is%20the%20body
```

Consider support those.

__UPD. Need real world examples, why/how such links can happen in human readable texts!__

### PR DESCRIPTION
**Pull request by melnik88**
I tried to fix #49 issue.

### CODE REVIEW

File: lib/re.js [hunk 1]
```diff
@@ -78,6 +78,10 @@ module.exports = function (opts) {
 
     '[\\-;:&=\\+\\$,\\"\\.a-zA-Z0-9_]+';
 
+  re.src_get_params =
```
Comment (puzrin) [inline lib/re.js]: Name is strange.

File: lib/re.js [hunk 2]
```diff
@@ -78,6 +78,10 @@ module.exports = function (opts) {
...
+
+    '((?:(?:\\?|\&)(?:[^=]+)\=(?:[^&]+))[^\\s\\n\\r\\)]+|)';
+
```
Comment (puzrin) [inline lib/re.js]: That's very suspicious.

- I would restrict param names with predefined values (subject|body|...). We should be sure that that false positive is not possible.
- It's good to break such equations to multiple strings with comments (if needed). Right now - very difficult to read.
- Not sure about condition when tail should be terminated.
- `[^\\s\\n\\r\\)]` hand-made !src_ZPCc ?

File: test/fixtures/links.txt [hunk 3]
```diff
@@ -259,7 +261,8 @@ email@example.com
 (email@example.com)
 email@example.com
 
-
+(someone@example.com?subject=This%20is%20the%20subject&cc=someone_else@example.com&body=This%20is%20the%20body)
+someone@example.com?subject=This%20is%20the%20subject&cc=someone_else@example.com&body=This%20is%20the%20body
```
Comment (puzrin) [inline test/fixtures/links.txt]: It worth to add more test cases. For example, with dot after email and so on.

Note, that false positive is more painful than unrecognized pattern.

**Comment by melnik88** [comment]
Ow, I see what you mean. 
Then I will close my MR. Thank you for commenting!

**Comment by puzrin** [comment]
I don't reject your work, it a good start. But it needs some care.

Also, if you have real data with such emails - post examples to #49 please (as much as possible). That can help to analyze required restrictions. I see no problems with correct percent-encoded links, but don't know if support for bad encoding required (as implemented  for http urls).

**Comment by melnik88** [comment]
I closed it not because of grievance but I like the idea of predefined link email params such as body, subject, cc, bcc (http://blog.escapecreative.com/customizing-mailto-links/).

Also i spied one disputable solution:
we highlight link and query params if scheme mailto exists but highlight only link in other case.

What do you think about it.

**Comment by puzrin** [comment]
Since this package use heuristics and it's 100% magical, my opinion is that real world examples should be collected first. Because universal solution is not possible.

**Comment by puzrin** [comment]
In worst case, you can add your modifications via `.onCompile()` or inheritance. I tried provide enougth flexibility to make users independent on my opinion.

**Pull request closed without merging**